src/sagemaker/session.py

```diff
@@ -4415,6 +4424,11 @@ def production_variant(
     if accelerator_type:
         production_variant_configuration["AcceleratorType"] = accelerator_type
 
+    if serverless_inference_config:
+        production_variant_configuration["ServerlessConfig"] = serverless_inference_config
+        production_variant_configuration.pop("InitialInstanceCount", None)
+        production_variant_configuration.pop("InstanceType", None)
```

Comment:
Oh great point. I did this to avoid changing existing code as `InitialInstanceCount` and `InstanceType` are required and set in original code. But let me change those two fields to be set only when `serverless_inference_config` is None. This will let us get rid of this pop thing and let the validation happens in CreateEndpointConfig request